engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -152,13 +151,19 @@ public static AnalyzerContext createContext(
             resultColumnNames.add(sc.getName());
 
             context.processedCols.add(sc);
+            if (sc.barriers() != null) {
+                preShiftBarriers.addAll(Arrays.asList(sc.barriers()));
+            }
         }
 
         compilationProcessor.compile();
 
+        final Map<Object, Integer> barrierToLayerIndex = new IdentityHashMap<>();
```

Comment:
If you were interested... `io.deephaven.util.datastructures.hash.IdentityKeyedObjectKey`